lib/p2p/Parser.ts

```diff
@@ -60,20 +60,30 @@ interface Parser {
 
 /** Protocol packet parser */
 class Parser extends EventEmitter {
+  private pending = '';
+
+  constructor(private delimiter: string) {
+    super();
+  }
+
   public feed = (data: Buffer): void => {
-    const dataStr = data.toString();
-    const packetLines = dataStr.split('\r\n');
-    packetLines.forEach((packetStr) => {
-      if (!packetStr) {
-        return;
-      }
-      try {
-        const packet = fromRaw(packetStr);
-        this.emit('packet', packet);
-      } catch (err) {
-        this.emit('error', err);
-      }
-    });
+    this.pending += data.toString();
```

Comment:
Suggest to protect the max size of this.pending to avoid accumulation of a very large message

Reply:
I was thinking of having this implemented later but maybe not too hard to take care of it here, what's a reasonable number? 4 MB? I can imagine the ORDERS packet getting fairly large.